File: 53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        vector <int> T (nums.size() + 1, 0);
        T[1] = nums[0]; //-INT_MAX => overflow
        int res = nums[0];
        for (int i = 2; i <= nums.size(); i++)
        {
            T[i] = T[i-1] + nums[i-1] > nums[i-1] ? T[i-1] + nums[i-1] : nums[i-1];
            res = T[i] > res ? T[i] : res;
        }
        return res;
    }
};

File: 53. Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        T = [0 for i in xrange(len(nums) + 1)]
        T[1], res = nums[0], nums[0]
        for i in xrange(2, len(nums)+1):
            T[i] = max(T[i-1]+nums[i-1], nums[i-1])
            res = max(res, T[i])
        return res
